queuelistener/listener_test.go

```diff
@@ -21,21 +21,20 @@ type testConnection struct {
 
 type testListener struct {
 	sync.Mutex
-	closed            bool
-	failNextTemporary bool
-	fail              bool
-	connsBeforeFail   int
-	addr              net.Addr
-	conns             chan *testConnection
+	closed          bool
+	failNextTimeout bool
+	fail            bool
+	connsBeforeFail int
+	addr            net.Addr
+	conns           chan *testConnection
 }
 
 type testError struct{}
 
-var errTemporary testError
+var errTimeout testError
 
-func (err testError) Error() string   { return "test error" }
-func (err testError) Timeout() bool   { return false }
-func (err testError) Temporary() bool { return true }
+func (err testError) Error() string { return "test error" }
+func (err testError) Timeout() bool { return false }
```

Comment:
Maybe the tests are waiting for Timeout() returning true?
I see that you wait for some log message but maybe it's the condition that is now different, because Temporary was "true" and Timeout is "false"